tool: sstable check should not require blob files

`cockroach debug pebble sstable check` panics when blob references are encountered, even though it never reads the values.
We should rework this command to just load blocks and verify checksums instead of scanning.

```
panic: blob references not configured [recovered, repanicked]

goroutine 1 [running]:
github.com/cockroachdb/cockroach/pkg/util/log/logcrash.RecoverAndReportPanic({0x8d296b0, 0xe3b7900}, 0xc00388e000)
  pkg/util/log/logcrash/crash_reporting.go:138 +0x6b
panic({0x7516ca0?, 0xc001602fc0?})
  GOROOT/src/runtime/panic.go:783 +0x132
github.com/cockroachdb/pebble/sstable.(*defaultInternalValueConstructor).GetInternalValueForPrefixAndValueHandle(0xc0088c8158, {0x71282d21a1dc?, 0xa?, 0x1?})
  external/com_github_cockroachdb_pebble/sstable/values.go:146 +0x6f3
github.com/cockroachdb/pebble/sstable/colblk.(*DataBlockIter).Next(0xc0088c8478)
  external/com_github_cockroachdb_pebble/sstable/colblk/data_block.go:1631 +0x342
github.com/cockroachdb/pebble/sstable.(*singleLevelIterator[...]).nextInternal(0x8ea4800, 0x0)
  external/com_github_cockroachdb_pebble/sstable/reader_iter_single_lvl.go:1551 +0x94
github.com/cockroachdb/pebble/sstable.(*singleLevelIterator[...]).Next(...)
  external/com_github_cockroachdb_pebble/sstable/reader_iter_single_lvl.go:1501
github.com/cockroachdb/pebble/tool.(*sstableT).runCheck.func1({_, _}, _, {0x7, 0x7e, 0x122, 0x0, 0x0, 0x0, 0x0, ...})
  external/com_github_cockroachdb_pebble/tool/sstable.go:199 +0x40f
github.com/cockroachdb/pebble/tool.(*sstableT).foreachSstable.func1({0x7fffc3779750, 0x1f}, 0xc0014383c0)
  external/com_github_cockroachdb_pebble/tool/sstable.go:565 +0x14e
github.com/cockroachdb/pebble/tool.processFiles[...].func1()
  external/com_github_cockroachdb_pebble/tool/util.go:379 +0x330
github.com/cockroachdb/pebble/tool.processFiles[...].func2()
  external/com_github_cockroachdb_pebble/tool/util.go:389 +0x110
github.com/cockroachdb/pebble/tool.walk({0x8cc01e0, 0xc00010c068}, {0x8dc9808, 0xe133950}, {0x7fffc3779750, 0x1f}, 0xc0037c9a98)
  external/com_github_cockroachdb_pebble/tool/util.go:332 +0x22b
github.com/cockroachdb/pebble/tool.processFiles[...]({0x8cc01e0, 0xc00010c068}, {0x8dc9808, 0xe133950}, {0xc001602130, 0x555e3c0?, 0xc0037c9b60?}, {0xc0037c9b68, 0x2, 0x2}, ...)
```

Jira issue: PEBBLE-1418

--- a/tool/sstable.go
+++ b/tool/sstable.go
@@ -178,18 +178,21 @@ func (s *sstableT) runCheck(cmd *cobra.Command, args []string) {
 	s.foreachSstable(stderr, args, func(path string, r *sstable.Reader, props sstable.Properties) {
 		fmt.Fprintf(stdout, "%s\n", path)
 
+		if err := r.ValidateBlockChecksums(); err != nil {
+			fmt.Fprintf(stdout, "  checksum validation failed: %s\n", err)
+		}
+
 		// Update the internal formatter if this comparator has one specified.
 		s.fmtKey.setForComparer(props.ComparerName, s.comparers)
 		s.fmtValue.setForComparer(props.ComparerName, s.comparers)
 
-		iter, err := r.NewIter(sstable.NoTransforms, nil, nil, sstable.AssertNoBlobHandles)
+		iter, err := r.NewIter(sstable.NoTransforms, nil, nil, sstable.DebugHandlesBlobContext)
 		if err != nil {
 			fmt.Fprintf(stderr, "%s\n", err)
 			return
 		}
 
-		// Verify that SeekPrefixGE can find every key in the table.
-		prefixIter, err := r.NewIter(sstable.NoTransforms, nil, nil, sstable.AssertNoBlobHandles)
+		prefixIter, err := r.NewIter(sstable.NoTransforms, nil, nil, sstable.DebugHandlesBlobContext)
 		if err != nil {
 			fmt.Fprintf(stderr, "%s\n", err)
 			return

--- a/tool/testdata/sstable_check
+++ b/tool/testdata/sstable_check
@@ -50,3 +50,8 @@ sstable check
 testdata/bad-magic-sst/000015.sst
 ----
 000015.sst: pebble/table: invalid table 000015: (bad magic number: 0xf6cff485b741e288)
+
+sstable check
+testdata/find-val-sep-db/000005.sst
+----
+000005.sst

